jib-core/src/main/java/com/google/cloud/tools/jib/frontend/SkaffoldFilesOutput.java

```diff
@@ -0,0 +1,110 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.frontend;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.google.cloud.tools.jib.blob.Blobs;
+import com.google.cloud.tools.jib.json.JsonTemplate;
+import com.google.cloud.tools.jib.json.JsonTemplateMapper;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Builds a JSON string containing files and directories that <a
+ * href="https://github.com/GoogleContainerTools/skaffold">Skaffold</a> should watch for changes
+ * (and consequently trigger rebuilds).
+ *
+ * <p>{@code buildFiles} consist of build definitions. Changes in these files indicate that the
+ * project structure may have changed, so Skaffold will refresh the file watch list when this
+ * happens.
+ *
+ * <p>{@code inputs} consist of source/resource directories. Skaffold will trigger a rebuild when
+ * changes are detected in these files.
+ *
+ * <p>{@code ignore} consists of files that the Skaffold file watcher should not watch.
+ *
+ * <p>Example:
+ *
+ * <pre>{@code
+ * {
+ *   "buildFiles": [
+ *     "buildFile1",
+ *     "buildFile2"
+ *   ],
+ *   "inputs": [
+ *     "src/main/java/",
+ *     "src/main/resources/"
+ *   ],
+ *   "ignore": [
+ *     "pathToIgnore"
+ *   ]
+ * }
+ * }</pre>
+ */
+public class SkaffoldFilesOutput {
+
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  private static class SkaffoldFilesTemplate implements JsonTemplate {
+
+    private final List<String> buildFiles = new ArrayList<>();
+
+    private final List<String> inputs = new ArrayList<>();
+
+    private final List<String> ignore = new ArrayList<>();
+  }
+
+  private final SkaffoldFilesTemplate skaffoldFilesTemplate = new SkaffoldFilesTemplate();
+
+  /**
+   * Adds a build file/directory.
+   *
+   * @param buildFile the path to the file
+   */
+  public void addBuildFile(Path buildFile) {
+    skaffoldFilesTemplate.buildFiles.add(buildFile.toString());
+  }
+
+  /**
+   * Adds an input file/directory.
+   *
+   * @param inputFile the path to the directory
+   */
+  public void addInput(Path inputFile) {
+    skaffoldFilesTemplate.inputs.add(inputFile.toString());
+  }
+
+  /**
+   * Adds an ignored file/directory.
+   *
+   * @param ignoreFile the path to the file
+   */
+  public void addIgnore(Path ignoreFile) {
+    skaffoldFilesTemplate.ignore.add(ignoreFile.toString());
+  }
+
+  /**
+   * Gets the added files in JSON format.
+   *
+   * @return the files in a JSON string
+   * @throws IOException if writing out the JSON fails
+   */
+  public String getJsonString() throws IOException {
+    return Blobs.writeToString(JsonTemplateMapper.toBlob(skaffoldFilesTemplate));
```

Comment:
Are we only using Blobs and JsonTemplateManager here? Would it make sense to use the underlying library to write out the string, and have this in plugins-common?

Reply:
We could for now, but I just wasn't sure if we wanted to use Blobs and JsonTemplateManager from outside of jib-core, since they're low-level/not part of the public api.

Reply:
Sorry I meant the underlying library we are using for json. With stuff from the jackson library (using `ObjectMapper`) directly.

Reply:
Hmm, yeah, I wanted to avoid duplicate code, but I guess it won't be as much as I thought.

Reply:
cool